### ISSUE TITLE
SDK fails to compile after updating to 2.12

### ISSUE THREAD

**Issue by nileshcse14** [DataDog/dd-sdk-ios#1878]
### Describe the bug

Swift package product 'OpenTelemetryApi' is linked as a static library by '' and 2 other targets. This will result in duplication of library code.

### Reproduction steps

Add Datdog in a package which has dependency to another package which also uses DataDog

### Affected SDK versions

17

### Latest working SDK version

17

### Did you confirm if the latest SDK version fixes the bug?

Yes

### Integration Methods

SPM

**Comment by ganeshnj** [comment]
Hi @nileshcse14,

Could you a minimal reproducible example app?

**Comment by nileshcse14** [comment]
So you have to create a Swift Package A which uses DataDog. then create another swift package B which uses Datadog and Package A both. Now once you add Package B in your project you will get the error

**Comment by nileshcse14** [comment]
[gh-1680.zip](https://github.com/user-attachments/files/15563151/gh-1680.zip)

**Comment by nileshcse14** [comment]
@ganeshnj  So we have to make our package as dynamic. Then the issue will come. I have modified the code and attached here. I think the issue is happening opentelemetry-swift library as its static type

**Comment by ganeshnj** [comment]
I can confirm this is due to the fact `OpenTelemetryApi` is being forced to be static. This forces it to be static linked to each target which is dynamic and finally results in duplication of symbols.

```
diff --git a/Package.swift b/Package.swift
index b914dac..799e140 100644
--- a/Package.swift
+++ b/Package.swift
@@ -11,7 +11,7 @@ let package = Package(
         .tvOS(.v11)
     ],
     products: [
-        .library(name: "OpenTelemetryApi", type: .static, targets: ["OpenTelemetryApi"]),
+        .library(name: "OpenTelemetryApi", targets: ["OpenTelemetryApi"]),
         .library(name: "OpenTelemetrySdk", type: .static, targets: ["OpenTelemetrySdk"]),
         .library(name: "ResourceExtension", type: .static, targets: ["ResourceExtension"]),
         .library(name: "URLSessionInstrumentation", type: .static, targets: ["URLSessionInstrumentation"]),
```

Ideally library writers shouldn't fix the library type so that SPM can decide based on their consumption ([also mentioned here](https://developer.apple.com/documentation/packagedescription/product/library(name:type:targets:)#parameters) but otel [choose to fix it to static](https://github.com/open-telemetry/opentelemetry-swift/pull/222).

I will open a discussion with otel folks to see if we can relax library type requirement (atleast for OpenTelemetryApi target).

**Comment by nileshcse14** [comment]
So we have multiple embedded targets in our App like Siri, entitlements. If swift package is static it creates linker issue. So i cant force my library to static

**Comment by ganeshnj** [comment]
In that case you could manually make sure the library doesn't get added two times.

The current setup is something like
```
app
|
package b
|       |
|       dd-sdk-ios
|
package a
|
dd-sdk-ios
|
OpenTelemetryApi
```

this can be simplified as following so only one copy of OpenTelemetryApi exists

```
app
|
package b
|
package a
|
dd-sdk-ios
|
OpenTelemetryApi
```


[simplified-tree.zip](https://github.com/user-attachments/files/15587100/simplified-tree.zip)

**Comment by nileshcse14** [comment]
@ganeshnj  I will try this. But it will be still a problem if the package is not owned by us. Will keep you posted .

**Comment by ganeshnj** [comment]
I raised this in otel-swift https://github.com/open-telemetry/opentelemetry-swift/issues/548

**Comment by maxep** [comment]
Solved in [2.14.0](https://github.com/DataDog/dd-sdk-ios/releases/tag/2.14.0)
